internal/cmd/skupper/site/site.go

```diff
@@ -4,12 +4,25 @@ Copyright © 2024 Skupper Team <skupper@googlegroups.com>
 package site
 
 import (
+	"fmt"
+	"github.com/skupperproject/skupper/api/types"
 	"github.com/skupperproject/skupper/internal/cmd/skupper/site/kube"
+	"github.com/skupperproject/skupper/internal/cmd/skupper/site/non_kube"
+	"github.com/skupperproject/skupper/pkg/config"
 	"github.com/spf13/cobra"
 )
 
 func NewCmdSite() *cobra.Command {
 
+	return NewCmdSiteFactory(config.GetPlatform())
```

Comment:
Good memory that I was asking about this @nluaces!

If I recall correctly, I think this is actually what I was wondering about in that line of questioning. Was worried about how testing and general organization would end up looking like specifically around this bit: where we're pulling a platform from some source of global mutable state from each and every command constructor rather than passing it in explicitly.

Reply:
If we pass the platform explicitly, I can create unit tests (pending for now) easily and the behavior of the constructor will be deterministic

Reply:
Definitely not a blocking suggestion! I think that redoing things that way may make our lives easier (at least for people like me who get distracted trying to sort out if/when/where that config.Platform variable was ever overwritten.)

It appears to me this probably doesn't exactly work correctly today with just the flag anyhow if I am reading cobra docs correctly.

Reply:
oh! you are correct, I need to take into account the flag as well, and I'm currently relying on this configuration that takes care of the environment variable. Definitely I need to give priority to the flag. Thank you for noticing!